spaceship/lib/spaceship/connect_api/models/user.rb

```diff
@@ -27,9 +29,15 @@ class User
         "allAppsVisible" => "all_apps_visible",
         "provisioningAllowed" => "provisioning_allowed",
         "emailVettingRequired" => "email_vetting_required",
-        "notifications" => "notifications"
+        "notifications" => "notifications",
+
+        "visibleApps" => "visible_apps"
       })
 
+      ESSENTIAL_INCLUDES = [
+        "visibleApps"
+      ].join(",")
```

Comment:
This isn't automatically used at all. You need to pass this to the parent API calls.

For example, you need to pass it to the corresponding methods as the default value of `includes` argument. This way App Store Connect API will include "visibleApps" objects within the response and spaceship client will automatically map it to `visible_apps` property. That's what `ESSENTIAL_INCLUDES` convention is supposed to work.

```ruby
def self.all(filter: {}, includes: Spaceship::ConnectAPI::User::ESSENTIAL_INCLUDES, limit: nil, sort: nil)
```


* https://github.com/fastlane/fastlane/pull/19053/files#diff-75baf27580ebd9af6509c2bbdb9fca7cd67b716bf144964ffc8e208c397d57bcR49
* https://github.com/fastlane/fastlane/pull/19053/files#diff-75baf27580ebd9af6509c2bbdb9fca7cd67b716bf144964ffc8e208c397d57bcR55
* https://github.com/fastlane/fastlane/blob/c0b94f1d5b38700b3b1b57031e9ec2931331ac5e/spaceship/lib/spaceship/connect_api/models/user.rb#L41-L45

Reply:
Otherwise `visible_apps` property ends up returning `nil` depending on how you get the `User` object.

Reply:
Thanks for the explanatory comment @ainame. 
I compared how `ESSENTIAL_INCLUDES` are used in `App`, `AppStoreVersion` models and I thought for a while if user `visible_apps` should be fetched by default and I wasn't sure if I should use them as default `inlcudes` argument. 
I added the definition though to allow to use it as in example `testing steps`, e.g.
```ruby
users = Spaceship::ConnectAPI::User.all(includes: Spaceship::ConnectAPI::User::ESSENTIAL_INCLUDES)
```
Otherwise those that would like to fetch it would have to investigate what value should be used in `includes`.

Should `visible_apps` be fetched by default in `self.find` and `self.all` methods?

Reply:
My point is that if it's not "by default", it shouldn't be named "ESSENTIAL". It sounds more like `OPTIONAL_INCLUDES` to me.
https://developer.apple.com/documentation/appstoreconnectapi/list_users

> Should visible_apps be fetched by default in self.find and self.all methods?

To me, it can be an "essential" info when it comes to handling users for ASC. Any advice to determine if it's essential? @joshdholtz

Reply:
If the advice will be to go with optional, I'll rename it to `OPTIONAL_INCLUDES`. Otherwise, I'll add essential as default values for includes parameters in `self.all` and `self.find`.

Reply:
Wow, I missed this conversation completely 😱 Sorry! I had my notifications under control for a while but lost control this past month 😬 But...

I think essential is fine here!